docs/core-concepts/router.md

```diff
@@ -0,0 +1,146 @@
+# Router
+The router is responsible for matching routes to urls and updating the browser history. It also provides a way to navigate to routes, access the current route, and provides some routing utilities. See [Router](/api/types/Router.md) type for more information.
+
+## Creating a Router
+Create a router using the `createRouter` utility.
+
+```ts {8}
+import { createRoute, createRouter } from '@kitbag/router'
+
+const routes = [
+  createRoute(...),
+  createRoute(...),
+] as const
+
+const router = createRouter(routes)
+```
+
+## Installing
+Once you have created a router you can install it into a Vue application using the `use` method.
+
+```ts {8}
+import { createRouter } from '@kitbag/router'
+import { createApp } from 'vue'
+import App from './App.vue'
+
+const router = createRouter(...)
+const app = createApp(App)
+
+app.use(router)
+app.mount('#app')
+```
+
+## Type Safety
+Kitbag Router utilizes [declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html) to provide the internal types to match the actual router you're using.
+
+```ts
+declare module '@kitbag/router' {
+  interface Register {
+    router: typeof router
+  }
+}
+```
+
+## Router Options
+There are several options you can pass to the router to customize its behavior. See the [RouterOptions](/api/types/RouterOptions.md) type for more information and a list of options.
+
+```ts
+const router = createRouter(routes, {
+  base: '/app',
```

Comment:
thoughts on expanding this example?

```suggestion
  base: '/app',
  initialUrl: '/home',
  historyMode: 'memory',
  prefetch: { components: true, props: false },
  rejections: { authNeeded: ... },
```

Reply:
I don't feel strongly, but I thought linking to the api docs for this would mean it would always be up to date and one less piece we have to manually manage.